enumerable_methods.rb

```diff
@@ -0,0 +1,127 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/CyclomaticComplexity
+# rubocop: disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    arr = self if self.class == Array
+    arr = flatten if self.class == Hash
+    arr = to_a if self.class == Range
+
+    count = 0
+    while count < arr.length
+      yield(arr[count], count)
+      count += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    my_each { |elem, count| yield(elem, count) }
+  end
+
+  def my_select
+    selected = []
+    my_each { |elem| selected << elem if yield(elem) }
+    selected
+  end
+
+  def my_all(arg = nil)
+    if block_given?
+      my_each { |elem| return false if yield(elem) != true }
+      return true
+    end
+
+    if arg.nil?
+      my_each { |elem| return false if elem.nil? || elem == false }
+    elsif arg.is_a? Class
+      my_each { |elem| return false if elem.class != arg }
+    elsif arg.class == Regexp
+      my_each { |elem| return false unless arg.match(elem) }
+    else
+      my_each { |elem| return false if elem != arg }
+    end
+    true
+  end
+
+  def my_any(arg = nil)
+    if block_given?
+      my_each { |elem| return true if yield(elem) == true }
+      return false
+    end
+
+    if arg.nil?
+      my_each { |elem| return true if elem.nil? || elem == true }
+    elsif arg.is_a? Class
+      my_each { |elem| return true if elem.class == arg }
+    elsif arg.class == Regexp
+      my_each { |elem| return true if arg.match(elem) }
+    else
+      my_each { |elem| return true if elem == arg }
+    end
+    false
+  end
```

Comment:
```rb
2.6.5 :040 > false_array = [nil, false, nil, false]
2.6.5 :041 > false_array.my_any
 => true 
2.6.5 :042 > false_array.any?
 => false 
2.6.5 :043 >
```

Reply:
Let us fix the `my_any` method, it is returning `NoMethodError` as in the screenshot below
![my_any](https://user-images.githubusercontent.com/46388113/89060251-3c42a100-d352-11ea-9da1-fdd903fb83e7.png)